src/runtime/minrpc/minrpc_logger.cc

```diff
@@ -0,0 +1,288 @@
+                                              int** tcodes, int* num_args) {
+  SetRPCCode(RPCCode::kCallFunc);
+  logger_->LogVal<void*>("call_handle: ", reinterpret_cast<void*>(*call_handle));
+  ret_handler_->GetHandleName(reinterpret_cast<void*>(*call_handle));
+  if (*num_args > 0) {
+    logger_->LogString(", ");
+  }
+  ProcessValues(values, tcodes, num_args);
+  next_->ExecNormalCallFunc(call_handle, values, tcodes, num_args);
+}
+
+void MinRPCExecuteWithLog::ExecCopyFromRemote(DLTensor** arr, uint64_t* num_bytes,
+                                              uint8_t** temp_data) {
+  SetRPCCode(RPCCode::kCopyFromRemote);
+  logger_->LogVal<void*>("data_handle: ", reinterpret_cast<void*>((*arr)->data));
+  logger_->LogDLDevice(", DLDevice(type,id):", &(*arr)->device);
+  logger_->LogVal<int64_t>(", ndim: ", (*arr)->ndim);
+  logger_->LogDLData(", DLDataType(code,bits,lane): ", &(*arr)->dtype);
+  logger_->LogVal<uint64_t>(", num_bytes:", *num_bytes);
+  next_->ExecCopyFromRemote(arr, num_bytes, temp_data);
+}
+
+int MinRPCExecuteWithLog::ExecCopyToRemote(DLTensor** arr, uint64_t* num_bytes,
+                                           uint8_t** data_ptr) {
+  SetRPCCode(RPCCode::kCopyToRemote);
+  logger_->LogVal<void*>("data_handle: ", reinterpret_cast<void*>((*arr)->data));
+  logger_->LogDLDevice(", DLDevice(type,id):", &(*arr)->device);
+  logger_->LogVal<int64_t>(", ndim: ", (*arr)->ndim);
+  logger_->LogDLData(", DLDataType(code,bits,lane): ", &(*arr)->dtype);
+  logger_->LogVal<uint64_t>(", byte_offset: ", (*arr)->byte_offset);
+  return next_->ExecCopyToRemote(arr, num_bytes, data_ptr);
+}
+
+void MinRPCExecuteWithLog::ExecSyscallFunc(RPCCode* code, TVMValue** values, int** tcodes,
+                                           int* num_args) {
+  SetRPCCode(*code);
+  if ((*code) == RPCCode::kFreeHandle) {
+    if (((*num_args) == 2) && ((*tcodes)[0] == kTVMOpaqueHandle) && ((*tcodes)[1] == kDLInt)) {
+      logger_->LogVal<void*>("handle: ", reinterpret_cast<void*>((*values)[0].v_handle));
+      if ((*values)[1].v_int64 == kTVMModuleHandle ||
+          (*values)[1].v_int64 == kTVMPackedFuncHandle) {
+        ret_handler_->ReleaseHandleName(reinterpret_cast<void*>((*values)[0].v_handle));
+      }
+    }
+  } else {
+    ProcessValues(values, tcodes, num_args);
+  }
+  next_->ExecSyscallFunc(code, values, tcodes, num_args);
+}
+
+void MinRPCExecuteWithLog::ThrowError(RPCServerStatus code, RPCCode info) {
+  logger_->LogString("-> Error\n");
+  next_->ThrowError(code, info);
+}
+
+void MinRPCExecuteWithLog::ProcessValues(TVMValue** values, int** tcodes, int* num_args) {
+  if (*tcodes != nullptr) {
+    logger_->LogString("[");
+    for (int i = 0; i < *num_args; ++i) {
+      logger_->LogTVMValue((*tcodes)[i], (*values)[i]);
+
+      if ((*tcodes)[i] == kTVMStr) {
+        if (strlen((*values)[i].v_str) > 0) {
+          ret_handler_->UpdateCurrHandleName((*values)[i].v_str);
```

Comment:
i think this is specific to `ModuleGetFunction` and `GetGlobalFunc` and `CallFunc`, right?

Reply:
As we talked, I keep the code as is for the moment (i.e. it adds all the strings in parameters to the handle name). In future we can change it once we had a better vision of what should or should not be logged.